src/main/java/one/util/streamex/AbstractStreamEx.java

```diff
@@ -163,6 +163,60 @@ public S filter(Predicate<? super T> predicate) {
     public <R> StreamEx<R> map(Function<? super T, ? extends R> mapper) {
         return new StreamEx<>(stream().map(mapper), context);
     }
+    /**
+     * Returns a stream consisting of the results of replacing each element of
+     * this stream with the contents of a mapped Stream produced by applying
+     * the provided mapping function to each element.
+     *
+     * <p>
+     * This is an <a href="package-summary.html#StreamOps">intermediate
+     * operation</a>.
+     *
+     * <p>
+     * The {@code mapAndFlatten()} operation has the effect of applying a
+     * one-to-many transformation to the elements of the stream, and then
+     * flattening the resulting elements into a new stream.
+     *
+     * @param <R> The element type of the new stream
+     * @param mapper a <a
+     *        href="package-summary.html#NonInterference">non-interfering </a>,
+     *        <a href="package-summary.html#Statelessness">stateless</a>
+     *        function to apply to each element which produces a
+     *        stream of new values
+     * @return the new stream
+     * @since 0.7.3
+     */
+    public <R> StreamEx<R> mapAndFlatten(Function<? super T, ? extends Stream<? extends R>> mapper) {
```

Comment:
I don't understand how it differs from the normal `flatMap`. Can you elaborate with an example? Thanks.

Reply:
Hi Tagir
You already helped me in the past and I must admit you know streams better than me.
That's why could be a silly PR ... result of my ignorance.

A lot of my code is mapping a `Stream<T>` to a `Stream<Stream<R>>` whereas I just need `Stream<R>`.
Therefore I tend to write a lot of 
```
stream// ==> Stream<T>
.map(mapper)//==>Stream<Stream<R>>
.flatMap(Function.identity())//Stream<R>
```

This function was created as shortcut to these lines

Reply:
But it's the same as `stream.flatMap(mapper)`!

Reply:
And were you using IntelliJ IDEA it would immediately tell you that :-)
![image](https://user-images.githubusercontent.com/5114450/80055976-42ec9b80-854d-11ea-8562-4a48566a697c.png)

Reply:
@amaembo (better laugh than cry) 
Me right now : 
![now](https://user-images.githubusercontent.com/28574442/80194751-228b1280-861b-11ea-9d60-fba8083880ed.gif)

Me using streams for years : 
![last5](https://user-images.githubusercontent.com/28574442/80194840-46e6ef00-861b-11ea-92d1-4e852eb37112.gif)

Me correcting my code for the next 5 years : 
![next5](https://user-images.githubusercontent.com/28574442/80194817-3df61d80-861b-11ea-820e-9db5fc4d1801.gif)